src/hotspot/share/opto/vectorization.cpp

```diff
@@ -1783,3 +1785,417 @@ void AlignmentSolver::trace_constrained_solution(const int C_const,
   }
 }
 #endif
+
+void VTransformGraph::add_vtnode(VTransformNode* vtnode) {
+  assert(vtnode->_idx == _vtnodes.length(), "position must match idx");
+  _vtnodes.push(vtnode);
+}
+
+// Compute a linearization of the graph. We do this with a reverse-post-order of a DFS.
+// This only works if the graph is a directed acyclic graph (DAG). The C2 graph, and
+// the VLoopDependencyGraph are both DAGs, but after introduction of vectors/packs, the
+// graph has additional constraints which can introduce cycles. Example:
+//
+//                                                       +--------+
+//  A -> X                                               |        v
+//                     Pack [A,B] and [X,Y]             [A,B]    [X,Y]
+//  Y -> B                                                 ^        |
+//                                                         +--------+
+//
+// We return "true" IFF we find no cycle, i.e. if the linearization succeeds.
+bool VTransformGraph::schedule() {
```

Comment:
Note: used to be `PacksetGraph::schedule`. Now not with topsort, but with reverse-post-order. This works with the pre/post visited sets, like in other places of C2.